sensor_test.py

```diff
@@ -0,0 +1,184 @@
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+
+    """
+    Test to set sensor reading interval to 1, get sensor info, validate that the sensor reading interval is set,
+    get sensor reading, wait for the interval, get sensor reading again, and validate that the readings are different.
+    """
+    # Step 1: Set sensor reading interval to 1
+    set_sensor_reading_interval(1)
+    
+    # Step 2: Get sensor info
+    sensor_info = get_sensor_info()
+    
+    # Step 3: Validate that sensor reading interval is set to interval from Step 1
+    assert sensor_info['reading_interval'] == 1, f"Sensor reading interval should be 1, but got {sensor_info['reading_interval']}"
+    
+    # Step 4: Get sensor reading
+    initial_reading = get_sensor_reading()
+    
+    # Step 5: Wait for interval specified in Step 1
+    # (Note: This step is simulated with a sleep of 1 second)
+    time.sleep(1)
+    
+    # Step 6: Get sensor reading
+    new_reading = get_sensor_reading()
+    
+    # Step 7: Validate that reading from Step 4 doesn't equal reading from Step 6
+    assert initial_reading != new_reading, "Sensor readings should be different after the interval"
+
+
+
+# Максимальна версія прошивки сенсора -- 15
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+import time
+
+# Максимальна версія прошивки сенсора -- 15
+max_firmware_version = 15
+
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
+    """
+    Test to update sensor firmware, validate firmware version increments, and ensure it doesn't update past the maximum version.
+    """
+    # Step 1: Get original sensor firmware version
+    original_version = get_sensor_info()['firmware_version']
+    
+    # Steps 2-5: Update firmware and validate version increment until max_firmware_version - 1
+    while original_version < max_firmware_version - 1:
+        # Step 2: Request firmware update
+        update_sensor_firmware()
+        
+        # Simulate time delay for firmware update
+        time.sleep(1)
```

Comment:
One second isn't enough to wait for the sensor to come back online after it's done updating. This is where we need to use the function for conditional wait we defined in the conftest.